### DISCUSSION TITLE
Allow fetching JSON refresh token through oAuth flow

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#8395] [Feature Requests]
As i am trying to get the directus_refresh_token via a jwt (for localhost development or mobile app) it is not possible for me to obtain it.

See:
https://github.com/directus/v8-archive/issues/1981
since i think that if the normal login can this, the sso auth will also allow this:
https://docs.directus.io/reference/api/system/authentication/#login

1. Call SSO Provider using https://mydomain.com/projectname/auth/sso/google?mode=jwt&redirect_url=https://myvueappdomain.com/callback

Now Step 2 is not working.
2. Getting https://myvueappdomain.com/callback?request_token=xxxxxxx. I got that token

**Comment by NilsBaumgartner1994** [comment]
I think in /src/controllers/auth.ts at line 332 it could either be implemented or is missing ?

**Comment by rijkvanzanten** [comment]
@NilsBaumgartner1994 How would you image the flow to work exactly when using it from your app? 

Seeing the API redirects to your own app, the only way to pass tokens around is in the query parameters. This isn't really safe, as those are easily viewable, accessible from JS, and often logged in access logs.

**Reply by NilsBaumgartner1994** [reply]
Yeah that is correct but currently i have this problem which i dont know how to properly configure the cors.

My Application (or app) is not running on the same host as directus. Therefore the cookie received from loggin in via an sso is set, but it is not send to the server (or is filtered)

REFRESH_TOKEN_COOKIE_SAME_SITE: 'lax'
CORS_ALLOWED_HEADERS: 'Accept,Accept-Encoding,Ac.....'

What am i missing here ?


Application: localhost
Directus: myother.domain.com

**Reply by rijkvanzanten** [reply]
> Therefore the cookie received from loggin in via an sso is set, but it is not send to the server (or is filtered)

How are you making requests back to the server? Are you including the cookies?

**Reply by NilsBaumgartner1994** [reply]
Yes i am including the cookies, but they are blocked by Chrome / Brave. This same problem seems to exist if you are not on the same origin as directus: https://github.com/axios/axios/issues/2303#issuecomment-679784028

**Reply by rijkvanzanten** [reply]
You should be able to explicitly allow your domain using `REFRESH_TOKEN_COOKIE_DOMAIN`. Cross domain cookies are getting a little finicky in browsers, but afaik Directus should have flags for all the important parts in the cookie and CORS to configure it in a way where the browser will take it 👍🏻

**Reply by NilsBaumgartner1994** [reply]
okay fixed it. But it still is open how to get an access_token on a native mobile application.

**Reply by NilsBaumgartner1994** [reply]
I think it would be great if a SSO login to Directus could have also a mode (cookie, json) (https://docs.directus.io/reference/api/system/authentication/#login)

**Reply by NilsBaumgartner1994** [reply]
Since otherwise i dont know how to optain the token. How are other applications this doing like amazon or google ?

**Reply by alihassan161820** [reply]
@NilsBaumgartner1994 did you find a solution for getting access_token in mobile app ?

**Reply by NilsBaumgartner1994** [reply]
Not a very pleasant one.

**Reply by NilsBaumgartner1994** [reply]
So the problem still somehow exists...

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
